File: apps/mobile/android/app/src/main/java/com/tchat/components/TchatSidebar.kt
package com.tchat.components

import androidx.compose.animation.*
import androidx.compose.animation.core.animateFloatAsState
import androidx.compose.animation.core.tween
import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.clickable
import androidx.compose.foundation.gestures.detectDragGestures
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.items
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.*
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.vector.ImageVector
import androidx.compose.ui.input.pointer.pointerInput
import androidx.compose.ui.platform.LocalDensity
import androidx.compose.ui.platform.LocalHapticFeedback
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.tooling.preview.Preview
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import androidx.compose.ui.zIndex
import com.tchat.designsystem.Colors
import com.tchat.designsystem.Spacing

/**
 * Sidebar navigation component following Tchat design system
 */
@Composable
fun TchatSidebar(
    isOpen: Boolean,
    onOpenChange: (Boolean) -> Unit,
    items: List<TchatSidebarItem>,
    modifier: Modifier = Modifier,
    selectedItem: String? = null,
    onSelectionChange: ((String?) -> Unit)? = null,
    mode: TchatSidebarMode = TchatSidebarMode.Overlay,
    position: TchatSidebarPosition = TchatSidebarPosition.Leading,
    size: TchatSidebarSize = TchatSidebarSize.Standard,
    showOverlay: Boolean = true,
    allowDismiss: Boolean = true,
    header: (@Composable () -> Unit)? = null,
    footer: (@Composable () -> Unit)? = null
) {
    val density = LocalDensity.current
    val hapticFeedback = LocalHapticFeedback.current

    var expandedSections by remember { mutableStateOf(setOf<String>()) }
    var dragOffset by remember { mutableStateOf(0f) }

    val sidebarWidth = with(density) { size.width.toPx() }

    val overlayAlpha by animateFloatAsState(
        targetValue = if (isOpen && mode == TchatSidebarMode.Overlay && showOverlay) 0.5f else 0f,
        animationSpec = tween(300),
        label = "overlay_alpha"
    )

    val sidebarOffset by animateFloatAsState(
        targetValue = when {
            mode == TchatSidebarMode.Permanent -> 0f
            isOpen -> 0f
            position == TchatSidebarPosition.Leading -> -sidebarWidth
            else -> sidebarWidth
        } + dragOffset,
        animationSpec = tween(300),
        label = "sidebar_offset"
    )

    Box(modifier = modifier.fillMaxSize()) {
        // Overlay
        if (mode == TchatSidebarMode.Overlay && showOverlay) {
            Box(
                modifier = Modifier
                    .fillMaxSize()
                    .background(Color.Black.copy(alpha = overlayAlpha))
                    .zIndex(1f)
                    .clickable(enabled = isOpen && allowDismiss) {
                        onOpenChange(false)
                    }
            )
        }

        // Sidebar
        Box(
            modifier = Modifier
                .fillMaxHeight()
                .width(size.width)
                .offset(x = with(density) { sidebarOffset.toDp() })
                .zIndex(2f)
                .then(
                    if (position == TchatSidebarPosition.Trailing) {
                        Modifier.align(Alignment.TopEnd)
                    } else {
                        Modifier.align(Alignment.TopStart)
                    }
                )
                .pointerInput(allowDismiss) {
                    if (allowDismiss) {
                        detectDragGestures(
                            onDragEnd = {
                                val threshold = sidebarWidth * 0.3f
                                val shouldClose = when (position) {
                                    TchatSidebarPosition.Leading -> dragOffset < -threshold
                                    TchatSidebarPosition.Trailing -> dragOffset > threshold
                                }

                                if (shouldClose) {
                                    onOpenChange(false)
                                }
                                dragOffset = 0f
                            }
                        ) { _, dragAmount ->
                            val maxDrag = sidebarWidth * 0.3f
                            dragOffset = when (position) {
                                TchatSidebarPosition.Leading -> maxOf(-maxDrag, minOf(0f, dragAmount.x))
                                TchatSidebarPosition.Trailing -> minOf(maxDrag, maxOf(0f, dragAmount.x))
                            }
                        }
                    }
                }
        ) {
            SidebarContent(
                items = items,
                selectedItem = selectedItem,
                expandedSections = expandedSections,
                onExpandedSectionsChange = { expandedSections = it },
                onSelectionChange = { itemId ->
                    onSelectionChange?.invoke(itemId)
                    hapticFeedback.performHapticFeedback(
                        androidx.compose.ui.hapticfeedback.HapticFeedbackType.LongPress
                    )
                    if (mode == TchatSidebarMode.Overlay) {
                        onOpenChange(false)
                    }
                },
                position = position,
                header = header,
                footer = footer
            )
        }
    }
}

@Composable
private fun SidebarContent(
    items: List<TchatSidebarItem>,
    selectedItem: String?,
    expandedSections: Set<String>,
    onExpandedSectionsChange: (Set<String>) -> Unit,
    onSelectionChange: (String?) -> Unit,
    position: TchatSidebarPosition,
    header: (@Composable () -> Unit)?,
    footer: (@Composable () -> Unit)?
) {
    Column(
        modifier = Modifier
            .fillMaxHeight()
            .background(Colors.background)
            .border(
                width = 1.dp,
                color = Colors.border,
                shape = RoundedCornerShape(0.dp)
            )
    ) {
        // Header
        header?.let { headerContent ->
            Box(
                modifier = Modifier.padding(Spacing.md)
            ) {
                headerContent()
            }
        }

        // Navigation Items
        LazyColumn(
            modifier = Modifier.weight(1f),
            verticalArrangement = Arrangement.spacedBy(0.dp),
            contentPadding = PaddingValues(vertical = Spacing.sm)
        ) {
            items(items) { item ->
                SidebarItemView(
                    item = item,
                    selectedItem = selectedItem,
                    expandedSections = expandedSections,
                    onExpandedSectionsChange = onExpandedSectionsChange,
                    onSelectionChange = onSelectionChange,
                    level = 0
                )
            }
        }

        // Footer
        footer?.let { footerContent ->
            Box(
                modifier = Modifier.padding(Spacing.md)
            ) {
                footerContent()
            }
        }
    }
}

@Composable
private fun SidebarItemView(
    item: TchatSidebarItem,
    selectedItem: String?,
    expandedSections: Set<String>,
    onExpandedSectionsChange: (Set<String>) -> Unit,
    onSelectionChange: (String?) -> Unit,
    level: Int
) {
    val hapticFeedback = LocalHapticFeedback.current
    val isSelected = selectedItem == item.id
    val isExpanded = expandedSections.contains(item.id)

    Column {
        // Main item
        Row(
            modifier = Modifier
                .fillMaxWidth()
                .clickable(enabled = !item.isDisabled) {
                    if (item.children.isNotEmpty()) {
                        val newExpanded = if (isExpanded) {
                            expandedSections - item.id
                        } else {
                            expandedSections + item.id
                        }
                        onExpandedSectionsChange(newExpanded)
                    } else {
                        onSelectionChange(item.id)
                        item.action?.invoke()
                    }
                    hapticFeedback.performHapticFeedback(
                        androidx.compose.ui.hapticfeedback.HapticFeedbackType.LongPress
                    )
                }
                .background(
                    if (isSelected) Colors.primary.copy(alpha = 0.1f) else Color.Transparent
                )
                .padding(
                    start = Spacing.md + (level * 16).dp,
                    end = Spacing.md,
                    top = Spacing.sm,
                    bottom = Spacing.sm
                ),
            verticalAlignment = Alignment.CenterVertically,
            horizontalArrangement = Arrangement.spacedBy(Spacing.sm)
        ) {
            // Icon
            item.icon?.let { icon ->
                Icon(
                    imageVector = icon,
                    contentDescription = null,
                    tint = when {
                        item.isDisabled -> Colors.textDisabled
                        isSelected -> Colors.primary
                        else -> Colors.textSecondary
                    },
                    modifier = Modifier.size(20.dp)
                )
            }

            // Title
            Text(
                text = item.title,
                fontSize = 14.sp,
                fontWeight = FontWeight.Medium,
                color = when {
                    item.isDisabled -> Colors.textDisabled
                    isSelected -> Colors.primary
                    else -> Colors.textPrimary
                },
                modifier = Modifier.weight(1f)
            )

            // Badge
            item.badge?.let { badge ->
                Text(
                    text = badge,
                    fontSize = 10.sp,
                    color = Colors.textOnPrimary,
                    modifier = Modifier
                        .background(
                            color = Colors.error,
                            shape = CircleShape
                        )
                        .padding(horizontal = 6.dp, vertical = 2.dp)
                )
            }

            // Expand/collapse indicator
            if (item.children.isNotEmpty()) {
                Icon(
                    imageVector = if (isExpanded) Icons.Default.ExpandLess else Icons.Default.ChevronRight,
                    contentDescription = null,
                    tint = Colors.textSecondary,
                    modifier = Modifier.size(16.dp)
                )
            }
        }

        // Children (if expanded)
        AnimatedVisibility(
            visible = isExpanded && item.children.isNotEmpty(),
            enter = fadeIn() + expandVertically(),
            exit = fadeOut() + shrinkVertically()
        ) {
            Column {
                item.children.forEach { child ->
                    SidebarItemView(
                        item = child,
                        selectedItem = selectedItem,
                        expandedSections = expandedSections,
                        onExpandedSectionsChange = onExpandedSectionsChange,
                        onSelectionChange = onSelectionChange,
                        level = level + 1
                    )
                }
            }
        }
    }
}

/**
 * Sidebar item data class
 */
data class TchatSidebarItem(
    val id: String,
    val title: String,
    val icon: ImageVector? = null,
    val badge: String? = null,
    val isDisabled: Boolean = false,
    val children: List<TchatSidebarItem> = emptyList(),
    val action: (() -> Unit)? = null
)

/**
 * Sidebar mode definitions
 */
enum class TchatSidebarMode {
    Overlay,
    Push,
    Permanent
}

/**
 * Sidebar position definitions
 */
enum class TchatSidebarPosition {
    Leading,
    Trailing
}

/**
 * Sidebar size definitions
 */
enum class TchatSidebarSize(
    val width: androidx.compose.ui.unit.Dp
) {
    Compact(240.dp),
    Standard(280.dp),
    Wide(320.dp)
}

// Preview
@Preview(showBackground = true)
@Composable
fun TchatSidebarPreview() {
    var isOpen by remember { mutableStateOf(true) }
    var selectedItem by remember { mutableStateOf<String?>("dashboard") }

    Box(modifier = Modifier.fillMaxSize()) {
        // Main content
        Column(
            modifier = Modifier
                .fillMaxSize()
                .background(Colors.surface)
                .padding(Spacing.lg),
            horizontalAlignment = Alignment.CenterHorizontally,
            verticalArrangement = Arrangement.Center
        ) {
            Text(
                text = "Main Content",
                fontSize = 24.sp,
                fontWeight = FontWeight.Bold,
                color = Colors.textPrimary
            )

            Spacer(modifier = Modifier.height(Spacing.md))

            Button(
                onClick = { isOpen = !isOpen }
            ) {
                Text("Toggle Sidebar")
            }
        }

        // Sidebar
        TchatSidebar(
            isOpen = isOpen,
            onOpenChange = { isOpen = it },
            selectedItem = selectedItem,
            onSelectionChange = { selectedItem = it },
            items = listOf(
                TchatSidebarItem(
                    id = "dashboard",
                    title = "Dashboard",
                    icon = Icons.Default.Dashboard
                ),
                TchatSidebarItem(
                    id = "messages",
                    title = "Messages",
                    icon = Icons.Default.Message,
                    badge = "5"
                ),
                TchatSidebarItem(
                    id = "projects",
                    title = "Projects",
                    icon = Icons.Default.Folder,
                    children = listOf(
                        TchatSidebarItem(
                            id = "active-projects",
                            title = "Active Projects",
                            icon = Icons.Default.Circle
                        ),
                        TchatSidebarItem(
                            id = "completed-projects",
                            title = "Completed",
                            icon = Icons.Default.CheckCircle
                        )
                    )
                ),
                TchatSidebarItem(
                    id = "team",
                    title = "Team",
                    icon = Icons.Default.People,
                    children = listOf(
                        TchatSidebarItem(
                            id = "members",
                            title = "Members",
                            icon = Icons.Default.Person
                        ),
                        TchatSidebarItem(
                            id = "roles",
                            title = "Roles & Permissions",
                            icon = Icons.Default.Key
                        )
                    )
                ),
                TchatSidebarItem(
                    id = "settings",
                    title = "Settings",
                    icon = Icons.Default.Settings
                )
            ),
            header = {
                Column(
                    verticalArrangement = Arrangement.spacedBy(Spacing.sm)
                ) {
                    Row(
                        horizontalArrangement = Arrangement.spacedBy(Spacing.sm),
                        verticalAlignment = Alignment.CenterVertically
                    ) {
                        Icon(
                            imageVector = Icons.Default.Apps,
                            contentDescription = null,
                            tint = Colors.primary,
                            modifier = Modifier.size(24.dp)
                        )

                        Text(
                            text = "Tchat",
                            fontSize = 18.sp,
                            fontWeight = FontWeight.Bold,
                            color = Colors.textPrimary
                        )
                    }

                    Divider()
                }
            },
            footer = {
                Column(
                    verticalArrangement = Arrangement.spacedBy(Spacing.sm)
                ) {
                    Divider()

                    Row(
                        horizontalArrangement = Arrangement.spacedBy(Spacing.sm),
                        verticalAlignment = Alignment.CenterVertically
                    ) {
                        Icon(
                            imageVector = Icons.Default.AccountCircle,
                            contentDescription = null,
                            tint = Colors.textSecondary,
                            modifier = Modifier.size(20.dp)
                        )

                        Column {
                            Text(
                                text = "John Doe",
                                fontSize = 12.sp,
                                fontWeight = FontWeight.Medium,
                                color = Colors.textPrimary
                            )

                            Text(
                                text = "john@example.com",
                                fontSize = 10.sp,
                                color = Colors.textSecondary
                            )
                        }
                    }
                }
            }
        )
    }
}

File: apps/mobile/ios/Sources/Components/TchatSidebar.swift
//
//  TchatSidebar.swift
//  TchatApp
//
//  Created by Claude on 22/09/2024.
//

import SwiftUI

/// Sidebar navigation component following Tchat design system
public struct TchatSidebar: View {

    // MARK: - Sidebar Types
    public enum SidebarMode {
        case overlay
        case push
        case permanent
    }

    public enum SidebarPosition {
        case leading
        case trailing
    }

    public enum SidebarSize {
        case compact
        case standard
        case wide
    }

    // MARK: - Sidebar Item
    public struct SidebarItem {
        let id: String
        let title: String
        let icon: String?
        let badge: String?
        let isDisabled: Bool
        let children: [SidebarItem]
        let action: (() -> Void)?

        public init(
            id: String,
            title: String,
            icon: String? = nil,
            badge: String? = nil,
            isDisabled: Bool = false,
            children: [SidebarItem] = [],
            action: (() -> Void)? = nil
        ) {
            self.id = id
            self.title = title
            self.icon = icon
            self.badge = badge
            self.isDisabled = isDisabled
            self.children = children
            self.action = action
        }
    }

    // MARK: - Properties
    @Binding private var isOpen: Bool
    @Binding private var selectedItem: String?
    @State private var expandedSections: Set<String> = []
    @State private var dragOffset: CGFloat = 0

    let items: [SidebarItem]
    let mode: SidebarMode
    let position: SidebarPosition
    let size: SidebarSize
    let showOverlay: Bool
    let allowDismiss: Bool
    let header: AnyView?
    let footer: AnyView?
    let onSelectionChange: ((String?) -> Void)?

    // MARK: - Private Properties
    private let colors = Colors()

    // MARK: - Computed Properties
    private var sidebarWidth: CGFloat {
        switch size {
        case .compact: return 240
        case .standard: return 280
        case .wide: return 320
        }
    }

    private var overlayOpacity: Double {
        guard mode == .overlay else { return 0 }
        return isOpen ? 0.5 : 0
    }

    private var sidebarOffset: CGFloat {
        let baseOffset: CGFloat

        switch mode {
        case .permanent:
            baseOffset = 0
        case .overlay, .push:
            if isOpen {
                baseOffset = 0
            } else {
                baseOffset = position == .leading ? -sidebarWidth : sidebarWidth
            }
        }

        return baseOffset + dragOffset
    }

    // MARK: - Initializer
    public init(
        isOpen: Binding<Bool>,
        selectedItem: Binding<String?> = .constant(nil),
        items: [SidebarItem],
        mode: SidebarMode = .overlay,
        position: SidebarPosition = .leading,
        size: SidebarSize = .standard,
        showOverlay: Bool = true,
        allowDismiss: Bool = true,
        header: AnyView? = nil,
        footer: AnyView? = nil,
        onSelectionChange: ((String?) -> Void)? = nil
    ) {
        self._isOpen = isOpen
        self._selectedItem = selectedItem
        self.items = items
        self.mode = mode
        self.position = position
        self.size = size
        self.showOverlay = showOverlay
        self.allowDismiss = allowDismiss
        self.header = header
        self.footer = footer
        self.onSelectionChange = onSelectionChange
    }

    // MARK: - Body
    public var body: some View {
        ZStack {
            // Overlay
            if mode == .overlay && showOverlay {
                Color.black
                    .opacity(overlayOpacity)
                    .ignoresSafeArea()
                    .onTapGesture {
                        if allowDismiss {
                            closeSidebar()
                        }
                    }
                    .animation(.easeInOut(duration: 0.3), value: isOpen)
            }

            // Sidebar
            HStack(spacing: 0) {
                if position == .trailing {
                    Spacer()
                }

                sidebarContent
                    .frame(width: sidebarWidth)
                    .offset(x: sidebarOffset)
                    .animation(.easeInOut(duration: 0.3), value: isOpen)
                    .gesture(
                        DragGesture()
                            .onChanged { value in
                                if allowDismiss {
                                    let translation = value.translation.x
                                    let maxDrag = sidebarWidth * 0.3

                                    if position == .leading {
                                        dragOffset = max(-maxDrag, min(0, translation))
                                    } else {
                                        dragOffset = min(maxDrag, max(0, translation))
                                    }
                                }
                            }
                            .onEnded { value in
                                let translation = value.translation.x
                                let threshold = sidebarWidth * 0.3

                                withAnimation(.easeInOut(duration: 0.3)) {
                                    if position == .leading && translation < -threshold {
                                        closeSidebar()
                                    } else if position == .trailing && translation > threshold {
                                        closeSidebar()
                                    }

                                    dragOffset = 0
                                }
                            }
                    )

                if position == .leading {
                    Spacer()
                }
            }
        }
    }

    // MARK: - Sidebar Content
    @ViewBuilder
    private var sidebarContent: some View {
        VStack(spacing: 0) {
            // Header
            if let header = header {
                header
                    .padding(.horizontal, Spacing.md)
                    .padding(.top, Spacing.md)
            }

            // Navigation Items
            ScrollView {
                LazyVStack(spacing: 0) {
                    ForEach(items, id: \.id) { item in
                        sidebarItemView(item, level: 0)
                    }
                }
                .padding(.vertical, Spacing.sm)
            }

            // Footer
            if let footer = footer {
                footer
                    .padding(.horizontal, Spacing.md)
                    .padding(.bottom, Spacing.md)
            }
        }
        .background(colors.background)
        .overlay(
            Rectangle()
                .fill(colors.border)
                .frame(width: 1),
            alignment: position == .leading ? .trailing : .leading
        )
    }

    // MARK: - Sidebar Item View
    @ViewBuilder
    private func sidebarItemView(_ item: SidebarItem, level: Int) -> some View {
        VStack(spacing: 0) {
            // Main item
            Button(action: {
                if !item.children.isEmpty {
                    toggleSection(item.id)
                } else {
                    selectItem(item.id)
                }
            }) {
                HStack(spacing: Spacing.sm) {
                    // Indentation
                    if level > 0 {
                        Rectangle()
                            .fill(Color.clear)
                            .frame(width: CGFloat(level * 16))
                    }

                    // Icon
                    if let icon = item.icon {
                        Image(systemName: icon)
                            .font(.system(size: 16))
                            .foregroundColor(itemIconColor(item))
                            .frame(width: 20)
                    }

                    // Title
                    Text(item.title)
                        .font(.system(size: 14, weight: .medium))
                        .foregroundColor(itemTextColor(item))
                        .frame(maxWidth: .infinity, alignment: .leading)

                    // Badge
                    if let badge = item.badge {
                        Text(badge)
                            .font(.caption2)
                            .foregroundColor(colors.textOnPrimary)
                            .padding(.horizontal, 6)
                            .padding(.vertical, 2)
                            .background(colors.error)
                            .clipShape(Capsule())
                    }

                    // Expand/collapse indicator
                    if !item.children.isEmpty {
                        Image(systemName: expandedSections.contains(item.id) ? "chevron.down" : "chevron.right")
                            .font(.system(size: 12))
                            .foregroundColor(colors.textSecondary)
                            .animation(.easeInOut(duration: 0.2), value: expandedSections.contains(item.id))
                    }
                }
                .padding(.horizontal, Spacing.md)
                .padding(.vertical, Spacing.sm)
                .background(itemBackground(item))
                .contentShape(Rectangle())
            }
            .disabled(item.isDisabled)
            .buttonStyle(PlainButtonStyle())

            // Children (if expanded)
            if !item.children.isEmpty && expandedSections.contains(item.id) {
                ForEach(item.children, id: \.id) { child in
                    sidebarItemView(child, level: level + 1)
                }
                .transition(.opacity.combined(with: .move(edge: .top)))
                .animation(.easeInOut(duration: 0.2), value: expandedSections)
            }
        }
    }

    // MARK: - Styling Methods
    private func itemBackground(_ item: SidebarItem) -> Color {
        if selectedItem == item.id {
            return colors.primary.opacity(0.1)
        }
        return Color.clear
    }

    private func itemTextColor(_ item: SidebarItem) -> Color {
        if item.isDisabled {
            return colors.textDisabled
        }
        if selectedItem == item.id {
            return colors.primary
        }
        return colors.textPrimary
    }

    private func itemIconColor(_ item: SidebarItem) -> Color {
        if item.isDisabled {
            return colors.textDisabled
        }
        if selectedItem == item.id {
            return colors.primary
        }
        return colors.textSecondary
    }

    // MARK: - Actions
    private func selectItem(_ itemId: String) {
        selectedItem = itemId
        onSelectionChange?(itemId)

        if let item = findItem(itemId) {
            item.action?()
        }

        // Haptic feedback
        let impactFeedback = UIImpactFeedbackGenerator(style: .light)
        impactFeedback.impactOccurred()

        // Close sidebar if in overlay mode
        if mode == .overlay {
            closeSidebar()
        }
    }

    private func toggleSection(_ sectionId: String) {
        withAnimation(.easeInOut(duration: 0.2)) {
            if expandedSections.contains(sectionId) {
                expandedSections.remove(sectionId)
            } else {
                expandedSections.insert(sectionId)
            }
        }

        // Haptic feedback
        let impactFeedback = UIImpactFeedbackGenerator(style: .light)
        impactFeedback.impactOccurred()
    }

    private func closeSidebar() {
        withAnimation(.easeInOut(duration: 0.3)) {
            isOpen = false
        }
    }

    private func findItem(_ itemId: String) -> SidebarItem? {
        func searchItems(_ items: [SidebarItem]) -> SidebarItem? {
            for item in items {
                if item.id == itemId {
                    return item
                }
                if let found = searchItems(item.children) {
                    return found
                }
            }
            return nil
        }

        return searchItems(items)
    }
}

// MARK: - Convenience Initializers
extension TchatSidebar {
    public init<Header: View, Footer: View>(
        isOpen: Binding<Bool>,
        selectedItem: Binding<String?> = .constant(nil),
        items: [SidebarItem],
        mode: SidebarMode = .overlay,
        position: SidebarPosition = .leading,
        size: SidebarSize = .standard,
        showOverlay: Bool = true,
        allowDismiss: Bool = true,
        @ViewBuilder header: () -> Header,
        @ViewBuilder footer: () -> Footer,
        onSelectionChange: ((String?) -> Void)? = nil
    ) {
        self.init(
            isOpen: isOpen,
            selectedItem: selectedItem,
            items: items,
            mode: mode,
            position: position,
            size: size,
            showOverlay: showOverlay,
            allowDismiss: allowDismiss,
            header: AnyView(header()),
            footer: AnyView(footer()),
            onSelectionChange: onSelectionChange
        )
    }
}

// MARK: - Preview
#if DEBUG
struct TchatSidebar_Previews: PreviewProvider {
    static var previews: some View {
        ZStack {
            Color(.systemGray6)
                .ignoresSafeArea()

            VStack {
                Text("Main Content")
                    .font(.title)
                    .foregroundColor(.primary)
            }

            TchatSidebar(
                isOpen: .constant(true),
                selectedItem: .constant("dashboard"),
                items: [
                    TchatSidebar.SidebarItem(
                        id: "dashboard",
                        title: "Dashboard",
                        icon: "chart.bar"
                    ),
                    TchatSidebar.SidebarItem(
                        id: "messages",
                        title: "Messages",
                        icon: "message",
                        badge: "5"
                    ),
                    TchatSidebar.SidebarItem(
                        id: "projects",
                        title: "Projects",
                        icon: "folder",
                        children: [
                            TchatSidebar.SidebarItem(
                                id: "active-projects",
                                title: "Active Projects",
                                icon: "circle.fill"
                            ),
                            TchatSidebar.SidebarItem(
                                id: "completed-projects",
                                title: "Completed",
                                icon: "checkmark.circle"
                            )
                        ]
                    ),
                    TchatSidebar.SidebarItem(
                        id: "team",
                        title: "Team",
                        icon: "person.2",
                        children: [
                            TchatSidebar.SidebarItem(
                                id: "members",
                                title: "Members",
                                icon: "person"
                            ),
                            TchatSidebar.SidebarItem(
                                id: "roles",
                                title: "Roles & Permissions",
                                icon: "key"
                            )
                        ]
                    ),
                    TchatSidebar.SidebarItem(
                        id: "settings",
                        title: "Settings",
                        icon: "gearshape"
                    )
                ],
                header: {
                    VStack(alignment: .leading, spacing: Spacing.sm) {
                        HStack {
                            Image(systemName: "app.badge")
                                .font(.title2)
                                .foregroundColor(.primary)

                            Text("Tchat")
                                .font(.headline)
                                .fontWeight(.bold)
                        }

                        Divider()
                    }
                },
                footer: {
                    VStack(spacing: Spacing.sm) {
                        Divider()

                        HStack {
                            Image(systemName: "person.circle")
                                .font(.title3)

                            VStack(alignment: .leading, spacing: 2) {
                                Text("John Doe")
                                    .font(.caption)
                                    .fontWeight(.medium)

                                Text("john@example.com")
                                    .font(.caption2)
                                    .foregroundColor(.secondary)
                            }

                            Spacer()
                        }
                    }
                }
            )
        }
        .previewLayout(.sizeThatFits)
    }
}
#endif